docs/tutorials/03_minimum_eigen_optimizer.ipynb

```diff
@@ -293,6 +297,270 @@
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Analysis of Samples\n",
+    "Objects of type `SolutionSample` (here denoted as *samples*) are a useful information source provided\n",
+    "by `OptimizationResult` objects. They contain information about input values `x`, corresponding (optimized) function\n",
+    "values `fval`, probability of obtaining that result `probability` and the solution status `status` (`SUCCESS`,\n",
+    "`FAILURE`, `INFEASIBLE`).\n",
+    "\n",
+    "They can be easily printed out in a better-readable format."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "outputs": [],
+   "source": [
+    "def print_samples(samples: List[SolutionSample],\n",
+    "                  max_len: int = None):\n",
+    "    status_str = {OptimizationResultStatus.SUCCESS: 'SUCCESS',\n",
+    "                  OptimizationResultStatus.FAILURE: 'FAILURE',\n",
+    "                  OptimizationResultStatus.INFEASIBLE: 'INFEASIBLE'}\n",
+    "    print('\\n'.join(['[{}]: fval: {:.4f}, probability: {:.2f} [{}]'.format(''.join(str(int(_)) for _ in e.x),\n",
+    "                                                                           e.fval,\n",
+    "                                                                           e.probability,\n",
+    "                                                                           status_str[e.status])\n",
+    "            for e in sorted(samples, key=lambda x: x.probability, reverse=True)][:max_len]))"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[010]: fval: -2.0000, probability: 0.12 [SUCCESS]\n",
+      "[000]: fval: 0.0000, probability: 0.12 [SUCCESS]\n",
+      "[110]: fval: 0.0000, probability: 0.12 [SUCCESS]\n",
+      "[100]: fval: 1.0000, probability: 0.12 [SUCCESS]\n",
+      "[001]: fval: 3.0000, probability: 0.12 [SUCCESS]\n",
+      "[101]: fval: 3.0000, probability: 0.12 [SUCCESS]\n",
+      "[011]: fval: 3.0000, probability: 0.12 [SUCCESS]\n",
+      "[111]: fval: 4.0000, probability: 0.12 [SUCCESS]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print_samples(qaoa_result.samples, max_len=10)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "We may also want to filter samples according to their status or to a certain percentile, i.e. with probability higher\n",
+    "than some threshold."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 50,
+   "outputs": [],
+   "source": [
+    "def get_filtered_samples(samples: List[SolutionSample],\n",
+    "                         percentile: float = 0,\n",
```

Comment:
`percentile` should be `threshold`. Here it does not compute percentile. It is just used as a threshold.